src/Unpacker.php

```diff
@@ -99,14 +99,59 @@ public function unpack(Operation $op, Result $result = null): Result
                     }
                 }
 
-                if (!$this->manipulator->addLink($package['dev'] ? 'require-dev' : 'require', $link->getTarget(), $constraint, $op->shouldSort())) {
-                    throw new \RuntimeException(sprintf('Unable to unpack package "%s".', $link->getTarget()));
+                $linkName = $link->getTarget();
+                $linkType = $package['dev'] ? 'require-dev' : 'require';
+
+                if (isset($links[$linkName])) {
+                    $links[$linkName]['constraints'][] = $constraint;
+                    if ('require' === $linkType) {
+                        $links[$linkName]['type'] = 'require';
+                    }
+                } else {
+                    $links[$linkName] = [
+                        'type' => $linkType,
+                        'name' => $linkName,
+                        'constraints' => [$constraint],
+                    ];
+                }
+            }
+        }
+
+        $jsonPath = Factory::getComposerFile();
+        $jsonContent = file_get_contents($jsonPath);
+        $jsonStored = json_decode($jsonContent, true);
+        $jsonManipulator = new JsonManipulator($jsonContent);
+
+        foreach ($links as $key => $link) {
+            // nothing to do, package is already present in the "require" section
+            if (isset($jsonStored['require'][$link['name']])) {
+                continue;
+            }
+
+            if (isset($jsonStored['require-dev'][$link['name']])) {
+                // nothing to do, package is already present in the "require-dev" section
+                if ('require-dev' === $link['type']) {
+                    continue;
                 }
+
+                // removes package from "require-dev", because it will be moved to "require"
+                // save stored constraint
+                $link['constraints'][] = $jsonStored['require-dev'][$link['name']];
+                $jsonManipulator->removeSubNode('require-dev', $link['name']);
+            }
+
+            $constraint = class_exists(Intervals::class, false)
+                ? Intervals::compactConstraint(MultiConstraint::create($link['constraints']))->setPrettyString()
```

Comment:
Shouldn't it be `getPrettyString()`? If yes, it probably means that we need a way to test Composer 2 along side Composer 1.

Reply:
We already test with composer 2 apparently, in .travis.yml:
`composer require --no-update composer/composer:^2`
So we might need a test case to cover this path?

Reply:
`getPrettyString()` fixed in 2ca067410379c2a3af8a23e4a48c789c01e83afc

even if CI runs with composer 2, the test present in this PR doesn't cover all use cases discussed, not sure the best path to enhance it: testing commands (require, remove, update, unpack) or testing the `Unpacker` class directly